modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceReleaser.java

```diff
@@ -188,17 +195,60 @@ private boolean hasDeclaredMethod(Class<?> clazz, String methodName, Class<?>...
    */
   private void shutdownMySqlAbandonedConnectionCleanupThread() {
 
-    Class<?> classAbandonedConnectionCleanupThread;
+    Class<?> classAbandonedConnectionCleanupThread = null;
     try {
-      classAbandonedConnectionCleanupThread =
-          this.getClass().getClassLoader().loadClass("com.mysql.jdbc.AbandonedConnectionCleanupThread");
+      for (String knownCleanupThreadClassAddress : CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES) {
+        try {
+          classAbandonedConnectionCleanupThread =
+              this.getClass().getClassLoader().loadClass(knownCleanupThreadClassAddress);
+        } catch (ClassNotFoundException e) {
+          logger.warn("No AbandonedConnectionCleanupThread registered with class address " + knownCleanupThreadClassAddress);
+          continue;
+        }
+      }
+      if (classAbandonedConnectionCleanupThread == null) {
+        logger.warn("No AbandonedConnectionCleanupThread class found. Check if thread class is between "
+            + CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES.toString());
+        throw new ClassNotFoundException(CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES.get(0));
+      }
       try {
         Method uncheckedShutdown = classAbandonedConnectionCleanupThread.getMethod("uncheckedShutdown");
         uncheckedShutdown.invoke(null);
       } catch (NoSuchMethodException e) {
         Method checkedShutdown = classAbandonedConnectionCleanupThread.getMethod("shutdown");
         checkedShutdown.invoke(null);
       }
+
+      try {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Extract this nested try block into a separate method. ([squid:S1141](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1141))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7685&issues=AWjzLS16otIXnmsUYkAs&open=AWjzLS16otIXnmsUYkAs)

[ISSUE_KEY]: AWjzLS16otIXnmsUYkAs